src/viur/core/modules/file.py

```diff
@@ -766,104 +799,121 @@ def download(self, blobKey: str, fileName: str = "", download: str = "", sig: st
             if "root" not in usr["access"] and "file-view" not in usr["access"]:
                 raise errors.Forbidden()
             validUntil = "-1"  # Prevent this from being cached down below
-            blob = bucket.get_blob(blobKey)
-            downloadFilename = ""
+            blob = GOOGLE_STORAGE_BUCKET.get_blob(blobKey)
+
         else:
             # We got an request including a signature (probably a guest or a user without file-view access)
             # First, validate the signature, otherwise we don't need to proceed any further
             if not self.hmac_verify(blobKey, sig):
                 raise errors.Forbidden()
             # Split the blobKey into the individual fields it should contain
             try:
-                dlPath, validUntil, downloadFilename = base64.urlsafe_b64decode(blobKey).decode("UTF-8").split("\0")
+                dlPath, validUntil, download_filename = base64.urlsafe_b64decode(blobKey).decode("UTF-8").split("\0")
             except:  # It's the old format, without an downloadFileName
                 dlPath, validUntil = base64.urlsafe_b64decode(blobKey).decode("UTF-8").split("\0")
-                downloadFilename = ""
+
             if validUntil != "0" and datetime.datetime.strptime(validUntil, "%Y%m%d%H%M") < datetime.datetime.now():
-                raise errors.Gone()
-            blob = bucket.get_blob(dlPath)
+                blob = None
+            else:
+                blob = GOOGLE_STORAGE_BUCKET.get_blob(dlPath)
+
         if not blob:
-            raise errors.Gone()
-        if downloadFilename:
-            contentDisposition = f"attachment; filename={downloadFilename}"
-        elif download:
-            fileName = self.sanitize_filename(blob.name.split("/")[-1])
-            contentDisposition = f"attachment; filename={fileName}"
-        else:
-            fileName = self.sanitize_filename(blob.name.split("/")[-1])
-            contentDisposition = f"filename={fileName}"
-        if isinstance(_credentials, ServiceAccountCredentials):
+            raise errors.Gone("The requested blob has expired.")
+
+        if not filename:
+            filename = download_filename or urlquote(blob.name.split("/")[-1])
+
+        content_disposition = "; ".join(
+            item for item in (
+                "attachment" if download else None,
+                f"filename={filename}" if filename else None,
+            ) if item
+        )
+
+        if isinstance(_CREDENTIALS, ServiceAccountCredentials):
             expiresAt = datetime.datetime.now() + datetime.timedelta(seconds=60)
```

Comment:
```suggestion
            if vaildUntil == "0":
                expiresAt =  datetime.timedelta(days=7)
            else:
                try:
                    expiresAt = datetime.datetime.strptime(vaildUntil, "%Y%m%d%H%M")
                    if expiresAt-datetime.datetime.now() > datetime.timedelta(days=7):
                        expiresAt = datetime.timedelta(days=7)
                except:
                    expiresAt = datetime.timedelta(seconds=60)
```

Reply:
Already talked to @ArneGudermann, but here for documentation: We'll gonna make this proposal in a separate PR apart of #1050 when #1050 is merged.